enabler/src/de/schildbach/pte/NegentweeProvider.java

```diff
@@ -0,0 +1,986 @@
+            JSONObject operator = leg.optJSONObject("operator");
+
+            LinkedList<Point> foundPoints = new LinkedList<>();
+
+            // First stop
+            Stop firstStop = stopFromJSONObject(stops.getJSONObject(0));
+            foundPoints.add(pointFromLocation(firstStop.location));
+
+            // Intermediate stops
+            LinkedList<Stop> foundStops = new LinkedList<>();
+            for (int j = 1; j < stops.length()-1; j++){
+                foundStops.add(stopFromJSONObject(stops.getJSONObject(j)));
+                foundPoints.add(pointFromLocation(foundStops.getLast().location));
+            }
+
+            // Last stop
+            Stop lastStop = stopFromJSONObject(stops.getJSONObject(stops.length() - 1));
+            foundPoints.add(pointFromLocation(lastStop.location));
+
+            switch (leg.getString("type").toLowerCase()) {
+                case "scheduled":
+                    Product lineProduct = productFromMode(mode.getString("type"), mode.getString("name"));
+
+                    StringBuilder legMessage = new StringBuilder();
+
+                    // Add attributes to leg message
+                    JSONArray legAttributes = leg.getJSONArray("attributes");
+                    for (int k = 0; k < legAttributes.length(); k++){
+                        JSONObject legAttribute = legAttributes.getJSONObject(k);
+
+                        if (legMessage.length() > 0)
+                            legMessage.append(", ");
+                        legMessage.append(WordUtils.capitalizeFirst(legAttribute.getString("title")));
+                    }
+
+                    // Add disturbances to leg message
+                    if (disturbances != null) {
+                        JSONArray legDisturbances = leg.getJSONArray("disturbancePlannerIds");
+                        for (int k = 0; k < legDisturbances.length(); k++) {
+                            String legDisturbanceId = legDisturbances.optString(k);
+
+                            if (legDisturbanceId != null && disturbances.containsKey(legDisturbanceId)) {
+                                JSONObject legDisturbance = disturbances.get(legDisturbanceId);
+
+                                if (legMessage.length() > 0)
+                                    legMessage.append("<br>\n<br>\n");
+                                legMessage.append(legDisturbance.getString("title"));
+                                legMessage.append(":<br>\n");
+                                legMessage.append(legDisturbance.getString("effect"));
+                                legMessage.append(" ");
+                                legMessage.append(legDisturbance.getString("measure"));
+                            }
+                        }
+                    }
+
+                    foundLegs.add(new Trip.Public(
+                            new Line(
+                                    leg.getString("service"), // final String id
```

Comment:
I'm not sure if these comments are useful. Your IDE should be able to tell the parameter names.